torch/distributed/distributed_c10d.py

```diff
@@ -335,12 +327,7 @@ def init_process_group(backend,
     backend = Backend(backend)
 
     if backend == Backend.MPI:
-        if not is_mpi_available():
-            raise RuntimeError("Distributed package doesn't have MPI built in")
-
-        _default_pg = ProcessGroupMPI([])
-        _pg_map[_default_pg] = (Backend.MPI, True)
-        _pg_names[_default_pg] = group_name
+        _default_pg = _new_process_group_helper(-1, -1, [], Backend.MPI, None)
```

Comment:
missing `group_name` here?

Reply:
Good catch -- though we should permanently remove it from the `init_process_group` call.

We only use it as prefix in a store, so that we can use it multiple times. It isn't exposed in the public API (e.g. to get or set the name) anywhere.

Reply:
Or, come to think of it, bring back multicast initialization, and use it to disambiguate multiple runs.

This is what is used to be used for...